### DISCUSSION TITLE
adding collaborator groups

### DISCUSSION THREAD

**Discussion by anjalijain22** [broadinstitute/seqr#4483] [Q&A]
Hello,

Hope you're doing well!

I am wondering how to add collaborator groups on seqr? Is this something that needs to be done through the admin page?
![Screenshot 2024-11-18 at 10 15 15 AM](https://github.com/user-attachments/assets/063d98f0-05e5-480f-a11c-ead67efe0772)

Many thanks,
Anjali

**Comment by hanars** [comment]
Yes, they need to be managed in the django admin

**Reply by anjalijain22** [reply]
Hi Hana,

Thanks for your prompt response! I have a few more follow up questions:

1. I'm trying to understand the purpose of creating user groups - is it to make the process of adding collaborators and managers more efficient, i.e., not having to add users one by one to a project? Is there another use case that I am missing?
2. While creating a user group, I see a bunch of available permissions (screenshot below). Which of these permission do you recommend adding while creating user groups?
3. On a related note, can these available permission be used to supplement permissions of other roles. For example, I have a "add_proj" user group with "Seqr | Project| Can add project" permission. I have a user X who is a data manager and thus inherently does not have the ability to create projects. If I add user X to the "add_proj" user group, would it mean they are now data managers with the ability to create projects on the front end too?

<img width="1175" alt="Screenshot 2024-11-18 at 12 02 28 PM" src="https://github.com/user-attachments/assets/06a3a0b4-94fb-4461-937f-5c0634002bf6">

Many thanks for your help, its greatly appreciated!

**Reply by hanars** [reply]
Hi Anjali, 
1) The purpose behind groups is if you have data in multiple projects where you always want the same users to have access, it is easier to create a group once and add it to your projects rather than add accounts individually. Also, if you have multiple projects and high turnover for your users it can be easier to remove someone who has left from a group once rather than remembering to remove them from all their projects. However if you do not have that many projects, or if the users who need acess differ between projects, there is no reason to use groups and you should just add users individually to projects. 
2) Do not add any permissions to your group when you create it, just users. Permissions will be handled when you add the group to the project from the main seqr app
3) I do not know what the `add_proj` permission is - the ability to create projects in seqr is determined by whether or not a user is a superuser, which is a property set directly on the user and not on a group or permission

**Reply by anjalijain22** [reply]
Thanks so much, Hana! That makes sense!

Re: 2. Is there an easier way to add/remove users from a group? The only way I can see right now is to go into admin/auth/users/, then select the user to edit and add them to the newly created group.

Re: 3. The `add_proj` user group has the "Seqr | Project| Can add project" permission. But from your message, I understand this wouldn't work in giving any user the ability to create projects unless they're a superuser

<img width="1123" alt="Screenshot 2024-11-18 at 12 40 14 PM" src="https://github.com/user-attachments/assets/b405c685-f126-4268-9143-adca74192c88">

**Reply by hanars** [reply]
There is not. If groups and access need to be edited frequently than groups are probably the wrong solution for you.
